src/components/EditProfile.jsx

```diff
@@ -0,0 +1,208 @@
+import React, { useContext, useEffect, useState } from "react";
+import { UserContext } from "../context/UserContext";
+import { userData } from "../UserData";
+import { useNavigate, useParams } from "react-router";
+
+const EditProfile = () => {
+  const param = useParams();
+  const navigate = useNavigate();
+  const { name } = useContext(UserContext);
+  const [formInputs, setFormInputs] = useState({
+    id: "",
+    userName: "",
+    firstName: "",
+    lastName: "",
+    designation: "",
+    favoriteFood: "",
+    hobbies: "",
+  });
+
+  const currentUser = userData.find((user) => user.name === param.name);
+
+  useEffect(() => {
+    setFormInputs(currentUser);
+  }, [currentUser]);
+
+  const handleChange = (e) => {
+    const { name, value } = e.target;
+
+    setFormInputs((prev) => {
+      return { ...prev, [name]: value };
+    });
+  };
+
+  const handleSubmit = (e) => {
+    e.preventDefault();
+    navigate(`/${name}`);
+  };
+
+  const deleteFood = (deletedFood) => {
+    const newFoodList = currentUser.favoriteFood
+      .split(",")
```

Comment:
Bhai, please create a variable first on splitting screen and then filter that array